sdcm/sct_events/event_handler.py

```diff
@@ -0,0 +1,67 @@
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation; either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright (c) 2022 ScyllaDB
+
+import logging
+import threading
+from functools import partial
+from typing import Tuple, Any, Optional
+
+from sdcm.cluster import TestConfig
+from sdcm.sct_events.events_processes import \
+    EVENTS_HANDLER_ID, EventsProcessesRegistry, BaseEventsProcess, \
+    start_events_process, get_events_process, verbose_suppress
+from sdcm.sct_events.handlers.schema_disagreement import SchemaDisagreementHandler
+
+LOGGER = logging.getLogger(__name__)
+
+
+class TestFailure(Exception):
+    pass
+
+
+class EventsHandler(BaseEventsProcess[Tuple[str, Any], None], threading.Thread):
+    """Runs handlers for events (according to EventsHandler.handlers mapping dict).
+
+     Handlers are created to gather additional information for issue investigation purposes."""
+    handlers = {
+        "CassandraStressLogEvent.SchemaDisagreement": SchemaDisagreementHandler(),
```

Comment:
won't it be better to reverse it ?, and that `SchemaDisagreementHandler` would be automaticlly register with this class, and hold the event data it's need for filtering ?

Reply:
Hmm... I don't quite get what you mean. 
Is it about making specific handler to update `EventsHandler` handlers map with some class method?

Reply:
I think that `SchemaDisagreementHandler` should be holding the information that it should be called for "CassandraStressLogEvent.SchemaDisagreement"

and only subclass from "EventHandler" should be enough.

anyhow it's not critical to start with, but if we'll implement more and more of those, it would be better a central registry for the event handlers.

we might want to be able to handle more then one event type (again without a specific usecase it's hard to tell exactly what we'll need more out of it)